include/deal.II/lac/trilinos_vector.h

```diff
@@ -1613,7 +1613,8 @@ namespace TrilinosWrappers
             {
               const int ierr = vector->ReplaceGlobalValues(
                 1,
-                (const TrilinosWrappers::types::int_type *)(&row),
+                reinterpret_cast<const TrilinosWrappers::types::int_type *>(
```

Comment:
Doesn't this suggest that we gave `row` above the wrong type?

Reply:
Yes, we are relying on `unsigned long long` having the same bit representation as `long long` for the values we are using here.

Reply:
I think I would prefer if we handle this through assignment -- change the data type above and then you don't have to use the reinterpret_cast here and below.